### ISSUE TITLE
Scaling not applied correctly when display is reconnected

### ISSUE THREAD

**Issue by lheckemann** [swaywm/sway#5245]
Otherwise, please include the following four components in your bug report: sway version, debug log, configuration (if applicable), and an explanation of steps taken to reproduce the issue. If sway crashes, also include a stack trace.

* Sway version:
e553e38270afa28ac7da8caf1d7e06890f476086

* Debug Log: https://gist.github.com/lheckemann/e4af2303ddf4a00acd9aa77f125b3cf3

* Configuration File: `/dev/null`

* Finally, explain the steps you took in plain English to reproduce the problem below.

1. Start sway with `sway -c /dev/null -d`
2. (from an SSH session with appropriate `SWAYSOCK`)
  ```
  swaymsg exec alacritty
  swaymsg move to output right
  swaymsg output DP-1 scale 2
  ```
3. Unplug external monitor
4. Plug external monitor back in

Expected: alacritty window renders at 2× scale both before and unplugging the monitor.
Observed: alacritty window renders at 2× scale as expected (pixels match display pixels) before unplugging the monitor. After plugging the monitor back in, the window seems to render in a smaller scale and be scaled up with interpolation.

**Comment by emersion** [comment]
Can you provide a `WAYLAND_DEBUG=1` log of your client?

```sh
WAYLAND_DEBUG=1 your-client >client.log 2>&1
# Then reproduce the bug
```

**Comment by lheckemann** [comment]
Updated the gist

**Comment by scossu** [comment]
I have a similar issue. When turning off the monitor, all GTK2 and QT applications display tiny, unreadable characters (not scaled up by interpolation). GTK3 is fine.

**Comment by scossu** [comment]
My [sway.log](https://gist.github.com/scossu/2c775b17414f17e297ae4c08e4d57c5a) obtained by: 

- `sway -d -c /dev /null 2>~/sway.log`
- Turn off monitor
- Turn monitor back on
- Exit sway from other tty

**Comment by emersion** [comment]
@scossu Your issue is an Xwayland bug, see the wiki.

**Comment by lheckemann** [comment]
I haven't had this issue in a long time, and @cherti it sounds like what you're encountering is a different problem, so I'm going to go ahead and close this -- maybe open a new issue with your debug logs?
